From fab336d25955910f99b083c3280c0abe7007d452 Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyich@gmail.com>
Date: Sun, 3 Dec 2023 23:06:30 +0000
Subject: [PATCH] test/mocklibc: fix build against

polkit> ../subprojects/mocklibc-1.0/src/netgroup-debug.c: In function 'netgroup_debug_print_entry':
polkit> ../subprojects/mocklibc-1.0/src/netgroup-debug.c:25:3: error: implicit declaration of function 'print_indent' [-Wimplicit-function-declaration]
polkit>    25 |   print_indent(stream, indent);
polkit>       |   ^~~~~~~~~~~~
---
 test/mocklibc/src/netgroup-debug.c | 11 +++++++++++
 test/mocklibc/src/netgroup.c       | 11 -----------
 2 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/test/mocklibc/src/netgroup-debug.c b/test/mocklibc/src/netgroup-debug.c
index 81d6e728..46e5b25f 100644
--- a/test/mocklibc/src/netgroup-debug.c
+++ b/test/mocklibc/src/netgroup-debug.c
@@ -21,6 +21,17 @@
 #include <stdio.h>
 #include <stdlib.h>
 
+/**
+ * Print a varaible indentation to the stream.
+ * @param stream Stream to print to
+ * @param indent Number of indents to use
+ */
+static void print_indent(FILE *stream, unsigned int indent) {
+  int i;
+  for (i = 0; i < indent; i++)
+    fprintf(stream, "  ");
+}
+
 void netgroup_debug_print_entry(struct entry *entry, FILE *stream, unsigned int indent) {
   print_indent(stream, indent);
 
diff --git a/test/mocklibc/src/netgroup.c b/test/mocklibc/src/netgroup.c
index 06a8a894..e16e4519 100644
--- a/test/mocklibc/src/netgroup.c
+++ b/test/mocklibc/src/netgroup.c
@@ -71,17 +71,6 @@ static char *parser_copy_word(char **cur) {
   return result;
 }
 
-/**
- * Print a varaible indentation to the stream.
- * @param stream Stream to print to
- * @param indent Number of indents to use
- */
-void print_indent(FILE *stream, unsigned int indent) {
-  int i;
-  for (i = 0; i < indent; i++)
-    fprintf(stream, "  ");
-}
-
 /**
  * Connect entries with 'child' type to their child entries.
  * @param headentry Head of list of entries that need to be connected
-- 
GitLab

